internal/telemetry/client_test.go

```diff
@@ -43,7 +47,13 @@ func TestClient(t *testing.T) {
 	client.Start(nil, nil) // test idempotence
 	defer client.Stop()
 
-	<-heartbeat
+	timeout := time.After(30 * time.Second)
```

Comment:
I think you have the best possible test here given what we're trying to test above :P I can't come up with anything better that doesn't have it's own set of problems like not actually testing all the code that this covers.

Reply:
Okay thanks for giving it another set of eyes!